### PR TITLE
ELEC-621: Add muxing across solar slaves for solar sense data messages

### PR DESCRIPTION
**Pull request by anikser**
Add signal multiplexing in dbc file for slave modules in SOLAR_DATA_FRONT and SOLAR_DATA_REAR.

### CODE REVIEW

**Comment by karlding** [comment]
Nice work! The DBC file changes look good.

```diff
diff --git 1/system_can.dbc 2/system_can_ali.dbc
index aa9f5ef..9813612 100644
--- 1/system_can.dbc
+++ 2/system_can_ali.dbc
@@ -268,16 +268,46 @@ BO_ 1411 DCDC_TEMPS: 4 CHAOS
  SG_ temp_1 : 0|16@1+ (1,0) [0|0] "" Vector__XXX
 
 BO_ 1450 SOLAR_DATA_FRONT: 8 SOLAR_MASTER_FRONT
- SG_ temperature : 48|16@1+ (1,0) [0|0] "" Vector__XXX
- SG_ current : 32|16@1+ (1,0) [0|0] "" Vector__XXX
- SG_ voltage : 16|16@1+ (1,0) [0|0] "" Vector__XXX
- SG_ module_id : 0|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_005 m5 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_004 m4 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_003 m3 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_002 m2 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_001 m1 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_000 m0 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_005 m5 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_004 m4 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_003 m3 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_002 m2 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_001 m1 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_000 m0 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_005 m5 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_004 m4 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_003 m3 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_002 m2 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_001 m1 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_000 m0 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ SOLAR_SLAVE_INDEX M : 0|16@1+ (1,0) [0|0] "" Vector__XXX
 
 BO_ 1483 SOLAR_DATA_REAR: 8 SOLAR_MASTER_REAR
- SG_ temperature : 48|16@1+ (1,0) [0|0] "" Vector__XXX
- SG_ current : 32|16@1+ (1,0) [0|0] "" Vector__XXX
- SG_ voltage : 16|16@1+ (1,0) [0|0] "" Vector__XXX
- SG_ module_id : 0|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_005 m5 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_004 m4 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_003 m3 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_002 m2 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_001 m1 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_TEMP_000 m0 : 48|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_005 m5 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_004 m4 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_003 m3 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_002 m2 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_001 m1 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_CURRENT_000 m0 : 32|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_005 m5 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_004 m4 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_003 m3 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_002 m2 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_001 m1 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ MODULE_VOLTAGE_000 m0 : 16|16@1+ (1,0) [0|0] "" Vector__XXX
+ SG_ SOLAR_SLAVE_INDEX M : 0|16@1+ (1,0) [0|0] "" Vector__XXX
 
 BO_ 1517 CHARGER_INFO: 6 CHARGER
  SG_ status_bitset : 32|16@1+ (1,0) [0|0] "" Vector__XXX

```

```
  Name:       SOLAR_DATA_FRONT
  Id:         0x5aa
  Length:     8 bytes
  Cycle time: - ms
  Senders:    SOLAR_MASTER_FRONT
  Layout:

                          Bit

             7   6   5   4   3   2   1   0
           +---+---+---+---+---+---+---+---+
         0 |------------------------------x|
           +---+---+---+---+---+---+---+---+
         1 |<------------------------------|
           +---+---+---+---+---+---+---+---+
             +-- SOLAR_SLAVE_INDEX
           +---+---+---+---+---+---+---+---+
         2 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
           +---+---+---+---+---+---+---+---+
         3 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
           +---+---+---+---+---+---+---+---+
             +-- MODULE_VOLTAGE_000
             +-- MODULE_VOLTAGE_001
             +-- MODULE_VOLTAGE_002
             +-- MODULE_VOLTAGE_003
             +-- MODULE_VOLTAGE_004
             +-- MODULE_VOLTAGE_005
     B     +---+---+---+---+---+---+---+---+
     y   4 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
     t     +---+---+---+---+---+---+---+---+
     e   5 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
           +---+---+---+---+---+---+---+---+
             +-- MODULE_CURRENT_000
             +-- MODULE_CURRENT_001
             +-- MODULE_CURRENT_002
             +-- MODULE_CURRENT_003
             +-- MODULE_CURRENT_004
             +-- MODULE_CURRENT_005
           +---+---+---+---+---+---+---+---+
         6 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
           +---+---+---+---+---+---+---+---+
         7 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
           +---+---+---+---+---+---+---+---+
             +-- MODULE_TEMP_000
             +-- MODULE_TEMP_001
             +-- MODULE_TEMP_002
             +-- MODULE_TEMP_003
             +-- MODULE_TEMP_004
             +-- MODULE_TEMP_005

  Signal tree:

    -- {root}
       +-- SOLAR_SLAVE_INDEX
           +-- 0
           |   +-- MODULE_VOLTAGE_000
           |   +-- MODULE_CURRENT_000
           |   +-- MODULE_TEMP_000
           +-- 1
           |   +-- MODULE_VOLTAGE_001
           |   +-- MODULE_CURRENT_001
           |   +-- MODULE_TEMP_001
           +-- 2
           |   +-- MODULE_VOLTAGE_002
           |   +-- MODULE_CURRENT_002
           |   +-- MODULE_TEMP_002
           +-- 3
           |   +-- MODULE_VOLTAGE_003
           |   +-- MODULE_CURRENT_003
           |   +-- MODULE_TEMP_003
           +-- 4
           |   +-- MODULE_VOLTAGE_004
           |   +-- MODULE_CURRENT_004
           |   +-- MODULE_TEMP_004
           +-- 5
               +-- MODULE_VOLTAGE_005
               +-- MODULE_CURRENT_005
               +-- MODULE_TEMP_005

  ------------------------------------------------------------------------

  Name:       SOLAR_DATA_REAR
  Id:         0x5cb
  Length:     8 bytes
  Cycle time: - ms
  Senders:    SOLAR_MASTER_REAR
  Layout:

                          Bit

             7   6   5   4   3   2   1   0
           +---+---+---+---+---+---+---+---+
         0 |------------------------------x|
           +---+---+---+---+---+---+---+---+
         1 |<------------------------------|
           +---+---+---+---+---+---+---+---+
             +-- SOLAR_SLAVE_INDEX
           +---+---+---+---+---+---+---+---+
         2 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
           +---+---+---+---+---+---+---+---+
         3 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
           +---+---+---+---+---+---+---+---+
             +-- MODULE_VOLTAGE_000
             +-- MODULE_VOLTAGE_001
             +-- MODULE_VOLTAGE_002
             +-- MODULE_VOLTAGE_003
             +-- MODULE_VOLTAGE_004
             +-- MODULE_VOLTAGE_005
     B     +---+---+---+---+---+---+---+---+
     y   4 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
     t     +---+---+---+---+---+---+---+---+
     e   5 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
           +---+---+---+---+---+---+---+---+
             +-- MODULE_CURRENT_000
             +-- MODULE_CURRENT_001
             +-- MODULE_CURRENT_002
             +-- MODULE_CURRENT_003
             +-- MODULE_CURRENT_004
             +-- MODULE_CURRENT_005
           +---+---+---+---+---+---+---+---+
         6 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
           +---+---+---+---+---+---+---+---+
         7 |XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX|
           +---+---+---+---+---+---+---+---+
             +-- MODULE_TEMP_000
             +-- MODULE_TEMP_001
             +-- MODULE_TEMP_002
             +-- MODULE_TEMP_003
             +-- MODULE_TEMP_004
             +-- MODULE_TEMP_005

  Signal tree:

    -- {root}
       +-- SOLAR_SLAVE_INDEX
           +-- 0
           |   +-- MODULE_VOLTAGE_000
           |   +-- MODULE_CURRENT_000
           |   +-- MODULE_TEMP_000
           +-- 1
           |   +-- MODULE_VOLTAGE_001
           |   +-- MODULE_CURRENT_001
           |   +-- MODULE_TEMP_001
           +-- 2
           |   +-- MODULE_VOLTAGE_002
           |   +-- MODULE_CURRENT_002
           |   +-- MODULE_TEMP_002
           +-- 3
           |   +-- MODULE_VOLTAGE_003
           |   +-- MODULE_CURRENT_003
           |   +-- MODULE_TEMP_003
           +-- 4
           |   +-- MODULE_VOLTAGE_004
           |   +-- MODULE_CURRENT_004
           |   +-- MODULE_TEMP_004
           +-- 5
               +-- MODULE_VOLTAGE_005
               +-- MODULE_CURRENT_005
               +-- MODULE_TEMP_005

  ------------------------------------------------------------------------
```

File: codegen/build_dbc.py [hunk 1]
```diff
@@ -194,19 +273,10 @@ def get_muxed_voltage_signal():
...
-                frame_id=frame_id,
-                name=can_frame.msg_name,
-                length=6,
-                signals=signals,
-                # The sender is the Message Source
-                senders=[can_frame.source]
-            )
-            database.messages.append(message)
+            total_length = 48
+        elif can_frame.msg_name == 'SOLAR_DATA_FRONT' or can_frame.msg_name == 'SOLAR_DATA_REAR':
```
Comment (karlding) [inline codegen/build_dbc.py]: Nit:

```python
can_frame.msg_name in ['SOLAR_DATA_FRONT', 'SOLAR_DATA_REAR']
```

**Review by karlding — DISMISSED** [review]
LGTM

**Review by karlding — APPROVED** [review]

**Comment by karlding** [comment]
Just FYI there's no need to force push since we use squash as our merge strategy. :)

**Comment by anikser** [comment]
Force of habit lol - thanks :)

**Pull request merged**